### ISSUE TITLE
Access to version from run-time information

### ISSUE THREAD

**Issue by QuAzI** [pyinstaller/pyinstaller#6433]
I have version generator which creates `version.rc` for windows and `version.py` for all.
If I build one file app and try to access `version.py` I see that he is excluded from build.
How to save `version.py` if I use it like
```python
    version_module = resource_path('version.py')
    if os.path.exists(version_module):
        spec = importlib.util.spec_from_file_location("version",
                                                      version_module)
        version = spec.loader.load_module()
        logger.debug(f'Version: {version.__version__}')
```

Or maybe I really don't need this file? How can I access to version data embedded into one file build?

**Comment by QuAzI** [comment]
Put version into __init__.py near application and then
```
import __init__

logger.debug(f'Version: {__init__.__version__}')
```
Crashes with `ModuleNotFoundError: No module named '__init__'`

**Comment by rokm** [comment]
You are expecting `version.py` to exist as a file, whereas PyInstaller collects pure-python modules as byte-compiled .pyc modules into executable-embedded archive. Although in this case, it is probably not collected at all due to indirect import. So if you want that code to work as is, you'll need to collect that `version.py` as a data file as well.

That said, you could also simplify the conditional import into
```python
try:
    import version
    logger.debug(f'Version: {version.__version__}')
except Exception:
    # No version module available
    pass
``` 
which should avoid the issue of indirect import and of missing source file altogether...

**Comment by bwoodsend** [comment]
As a side note, you should never be importing `__init__`. That init file should belong to a package and it's that package you should be importing.

**Comment by QuAzI** [comment]
Looks like better always have `version.py` but don't store any data in it or store minimal default data and append it on build

**Comment by QuAzI** [comment]
Ok. Can I use some py file to completely replace `version.rc`? With author, company, dates

**Comment by rokm** [comment]
Within your program, you can organize and display version (and company, dates, etc.) information however you want.

But if you want that information to be part of executable and visible to the OS itself, you'll need to use `VERSIONINFO`.

**Comment by QuAzI** [comment]
Solved. I made `version.py` which contains all data to generate `version.rc` on the fly and can be embedded into CI.
[Short demo](https://github.com/QuAzI/py-version) where GitHub Actions gets data from version.py and replaces build number to set actual tag, release version and each other.

And current [version.py.zip](https://github.com/pyinstaller/pyinstaller/files/7711724/version.zip). Hope it can help somebody
